lib/language_pack/rails51.rb

```diff
@@ -0,0 +1,85 @@
+require 'securerandom'
+require "language_pack"
+require "language_pack/rails5"
+
+class LanguagePack::Rails51 < LanguagePack::Rails5
+  NODE_MODULES_PATH = 'node_modules'
+  WEBPACKER_PACKS_PATH = 'public/packs'
+  WEBPACKER_CACHE_PATH = 'tmp/cache/webpacker'
+  YARN_CACHE_PATH = '~/.yarn-cache'
+
+  # @return [Boolean] true if it's a Rails 5.1.x app
+  def self.use?
+    instrument "rails51.use" do
+      rails_version = bundler.gem_version('railties')
+      return false unless rails_version
+      is_rails = rails_version >= Gem::Version.new('5.1.x')
```

Comment:
I'd make this more explicit. This will never get used because the Rails 6 language pack will be tested for first. As is, this package appears after Rails6 in the `packs` list. I suspect the buildpack will defer to that and never call this code.

TBH, I think you'd be better off moving these  changes _into_ the Rails 6 pack. Apart from making it clearer, you are also missing stuff (the creation of the `tmp/pids` folder) than are specific to the Rails 6 install.